### ISSUE TITLE
(Regression) NuGet Layout Still Broken

### ISSUE THREAD

**Issue by jamiehankins** [dotnet/docfx#7570]
**Operating System**: Windows

**DocFX Version Used**: 2.58.2 

**Template used**: default

**Steps to Reproduce**: 

1. Create a C# project with a package reference to DocFX, like this:
```XML
<PackageReference
  Include="docfx.console"
  Version="2.58.2"
  PrivateAssets="all"
  IncludeAssets="runtime; build; native; contentfiles; analyzers; buildtransitive" />
```
2. Configure a DocFX build.
3. Build

**Expected Behavior**:
docfx.exe is run

**Actual Behavior**:
docfx.exe is not run

This was originally filed as https://github.com/dotnet/docfx/issues/7557.

The PR to fix it, https://github.com/dotnet/docfx/pull/7560, pinned version 5.10 of NuGet. Unfortunately, according to https://github.com/NuGet/Home/issues/11125, the regression in NuGet was introduced in 5.10. **The version that should be pinned is 5.9.1.**

**Comment by jamiehankins** [comment]
Hi @yufeih any chance of 2.58.3 for this? We've been waiting for the the older fixes (https://github.com/dotnet/docfx/issues/7422) for a long time, and this (hopefully) is the last thing blocking us.

**Comment by tibel** [comment]
Thee issue is not fixed in 2.58.3

**Comment by jamiehankins** [comment]
Confirmed that it's not fixed in 2.58.3.

The fix above by @v-pegao  looks promising. Hopefully, 2.58.4 will be the end of this issue.

**Comment by yufeih** [comment]
Fixed in 2.58.4
